src/NRules/NRules/Utilities/ExpressionComparer.cs

```diff
@@ -104,6 +118,26 @@ private static bool ExpressionEqual(Expression x, Expression y, LambdaExpression
             throw new NotImplementedException(x.ToString());
         }
 
+        private static bool MemberBindingCollectionsEqual(IEnumerable<MemberBinding> x, IEnumerable<MemberBinding> y,
+            LambdaExpression rootX, LambdaExpression rootY)
+        {
+            return x.Count() == y.Count()
+                   && x.Select((b, i) => new { Binding = b, Index = i })
+                       .Join(y.Select((b, i) => new { Binding = b, Index = i }),
+                           o => o.Index, o => o.Index, (xb, yb) => new { X = xb.Binding, Y = yb.Binding })
+                       .All(o => MemberBindingsEqual(o.X, o.Y, rootX, rootY));
```

Comment:
Since the collections are of the same size and joined by index, I think it's better to use `Zip` instead of `Join`.
```c#
            return x.Count() == y.Count()
                   && x.Zip(y, (a, b) => new { A = a, B = b })
                       .All(o => MemberBindingsEqual(o.A, o.B, rootX, rootY));
```

Reply:
Should i change CollectionsEqual to `Zip` as well?

Reply:
I changed all to Zip